pkg/infoschema/infoschema.go

```diff
@@ -320,6 +320,29 @@ func (is *infoSchema) SchemaTableInfos(schema model.CIStr) []*model.TableInfo {
 	return getTableInfoList(is.SchemaTables(schema))
 }
 
+type tableInfoResult struct {
+	DBName    string
+	TableInfo []*model.TableInfo
+}
+
+func (is *infoSchema) ListTablesWithSpecialAttribute(filter specialAttributeFilter) <-chan tableInfoResult {
```

Comment:
```suggestion
func (is *infoSchema) ListTablesWithSpecialAttribute(dbName string, filter specialAttributeFilter) <-chan tableInfoResult {
```
can we change it to get all tables with matched attributes for a db? so we don't to have a routine to do it. it's more complex now

Reply:
This makes the streaming API possible. For example, if we don't store the special attribute and need to fetch data from tikv, then return all data []XXX may use much more memory then this API setting. @D3Hunter